### ISSUE TITLE
Support JAX 0.2.5

### ISSUE THREAD

**Issue by f0uriest** [ddudt/DESC#44]
they changed some stuff in the latest version, so our code needs to change as well. Did some testing and looks like the major stuff is how we're using ``jnp.where`` and ``put`` to do logical indexing and stuff. Updating that should also fix ``NaN``s in reverse mode autodiff when including a point at the axis. 
Looks like there's also some stuff with the datatypes for static arguments, still trying to figure that out.
